google/cloud/forseti/scanner/scanners/ke_version_scanner.py

```diff
@@ -12,141 +12,20 @@
-                'rule_index': violation.rule_index,
-                'rule_name': violation.rule_name,
-                'violation_type': violation.violation_type,
-                'violation_data': violation_data,
-                'resource_data': violation.resource_data
-            }
-
-    def _output_results(self, all_violations):
-        """Output results.
-
-        Args:
-            all_violations (list): All violations
-        """
-        all_violations = list(self._flatten_violations(all_violations))
-        self._output_results_to_db(all_violations)
-
-    def _find_violations(self, ke_clusters):
-        """Find violations in the policies.
-
-        Args:
-            ke_clusters (list): Clusters to find violations in.
-
-        Returns:
-            list: All violations.
-        """
-        all_violations = []
-        LOGGER.info('Finding ke cluster version violations...')
-
-        for ke_cluster in ke_clusters:
-            violations = self.rules_engine.find_policy_violations(
-                ke_cluster)
-            LOGGER.debug(violations)
-            all_violations.extend(violations)
-        return all_violations
-
-    def _retrieve(self):
-        """Runs the data collection.
-
-        Returns:
-            list: KE Cluster data.
-        """
-        model_manager = self.service_config.model_manager
-        scoped_session, data_access = model_manager.get(self.model_name)
-        with scoped_session as session:
-            ke_clusters = []
-            for ke_cluster in data_access.scanner_iter(
-                    session, 'kubernetes_cluster'):
-                project_id = ke_cluster.parent.name
-                ke_clusters.append(
-                    KeCluster.from_json(project_id,
-                                        None,
-                                        ke_cluster.data,
-                                        ke_cluster.full_name))
-
-        # Retrieve the service config via a separate query because session
-        # in the middle of yield_per() can not support simultaneous queries.
-        with scoped_session as session:
-            for ke_cluster in ke_clusters:
-                position = (
-                    ke_cluster.resource_full_name.find('kubernetes_cluster'))
-                ke_cluster_type_name = (
-                    ke_cluster.resource_full_name[position:][:-1])
-
-                service_config = list(data_access.scanner_iter(
-                    session, 'kubernetes_service_config',
-                    parent_type_name=ke_cluster_type_name))[0]
-                ke_cluster.server_config = json.loads(service_config.data)
-
-        return ke_clusters
-
-    def run(self):
-        """Run, the entry point for this scanner."""
-        ke_clusters = self._retrieve()
-        all_violations = self._find_violations(ke_clusters)
-        self._output_results(all_violations)
+KeVersionScanner = ke_base_scanner.ke_scanner_factory(
```

Comment:
Nit:

I changed my mind on this a bit after working on https://github.com/GoogleCloudPlatform/forseti-security/pull/2105/files.

I would prefer to make KeVersionScanner explicitly a class:

class KeVersionScanner(ke_base_scanner.ke_scanner_factory('version', ke_version_rules_engine.KeVersionRulesEngine)):
    """Scanner class for KE version rules."""

Same for KeScanner in ke_scanner.py.

Reply:
no problem!